### DISCUSSION TITLE
Share installed extensions

### DISCUSSION THREAD

**Discussion by Angelk90** [zed-industries/zed#22071] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Share the installed extensions with other users, so that they too can install them if they want, importing them via a file.

Example there are A and B.

A has installed some extensions on his machine, let's consider that he has twenty installed.
To avoid having to write all the names of the extensions one by one, that he has installed on his IDE zed.
Just export a file that contains the names of the extensions, example json.

B imports this file, when the file is imported a window/modal appears something similar that through a checkbox makes you select the extensions that you want to install (understanding if some of them are already installed it highlights it), then after selecting the extensions through a button they are installed.

### Environment

```
Zed: v0.164.0 (Zed Preview)
OS: macOS 14.1.1
Memory: 16 GiB
Architecture: x86_64
```

**Comment by notpeter** [comment]
This is similar the `.vscode/extensions.json` functionality where a project can "recommend" certain extensions by id.

**Comment by notpeter** [comment]
I have concerns about the UX of such a feature.  I **hate** when VSCode triggers a bunch of pop ups as soon as I open a project.  Similarly, the low uptake of `extensions.json` / need for yet another config file in every project gives me pause.

**Reply by fksune** [reply]
I understand this can be annoying for some people, but sometimes it is a necessary feature to maintain project consistency. so may be put this behind a toggle?

**Reply by andre-morise** [reply]
@notpeter  I get the concern about popups, but I think there’s a good middle ground that fits Zed’s design. Instead of showing prompts when opening a project, Zed could just indicate workspace-recommended extensions directly inside the Zed Extension Gallery. 

For example, if a project includes a `.zed/extensions.json` file like:

```json
{
  "recommendations": [
    "toml",
    "tombi",
    "git-firefly",
    "markdown-oxide"
    "log",
  ]
}
```
As stated prior, this format mirrors VS Code’s `.vscode/extensions.json`, which has proven useful for maintaining consistent environments across teams. Alternatively, Zed could support a TOML variant.

In the Extensions Gallery, these could show a “Workspace Recommended” label (new `ExtensionCategoryFilter` + `ExtensionProvides::WorkspaceRecommended`) and/or appear at the top of the list, without any popups.

A new optional setting could be integrated into existing structures, to control visibility of this feature.
[crates/settings/src/settings_content/extension.rs](https://github.com/zed-industries/zed/blob/main/crates/settings/src/settings_content/extension.rs)
```rs
pub struct ExtensionSettingsContent {
    /// Whether to display workspace extension recommendations in the Extension Gallery.
    /// 
    /// Default: false
    #[serde(default)]
    pub show_recommended_extensions: Option<bool>, 
    // or perhaps more specifically, incase of confusion with built-in extension recommendations:
    // show_workspace_recommended_extensions
}
```
[crates/extension_host/src/extension_settings.rs](https://github.com/zed-industries/zed/blob/main/crates/extension_host/src/extension_settings.rs)
```rs
#[derive(Debug, Default, Clone)]
pub struct ExtensionSettings {
    /// Whether to display workspace extension recommendations in the Extension Gallery.
    /// 
    /// Default: false
    pub show_recommended_extensions: Option<bool>, 
    // or perhaps more specifically, incase of confusion with built-in extension recommendations:
    // show_workspace_recommended_extensions
}
```
_... plus any relevant modifications to [crates/extension_host/src/extension_host.rs](https://github.com/zed-industries/zed/blob/main/crates/extension_host/src/extension_host.rs) and [crates/extensions_ui/src/extensions_ui.rs](https://github.com/zed-industries/zed/blob/main/crates/extensions_ui/src/extensions_ui.rs)._

With the default set to `false` ([assets/settings/default.json](https://github.com/zed-industries/zed/blob/main/assets/settings/default.json)), users who prefer a minimal UI wouldn’t see recommendations unless they enable the feature in their `settings.json`. That being said, with the default set to `true`, I believe that the positive impact on the UX would outweigh any negatives.

This would make it easy for teams to stay consistent with shared tools and language servers while helping new contributors get started quickly. It keeps things discoverable, non-intrusive, and hopefully aligns well with Zed’s design. Though this may be niche, it provides nice "quality of life" for those that utilize this in other editors.

**Comment by rgeronimi** [comment]
This is really a needed feature once industrializing a little bit the usage of zed in dev teams (standard configuration, backups, ...), any plan? Thx
